src/QMCWaveFunctions/TWFFastDerivWrapper.cpp

```diff
@@ -921,4 +922,37 @@ TWFFastDerivWrapper::IndexType TWFFastDerivWrapper::getRowM(const ParticleSet& P
   return sid;
 }
 
+void TWFFastDerivWrapper::createResource(ResourceCollection& collection)
+{
+  collection.addResource(std::make_unique<TWFFastDerivWrapperMultiWalkerMem>());
+}
+
+void TWFFastDerivWrapper::acquireResource(ResourceCollection& collection,
+                                          const RefVectorWithLeader<TWFFastDerivWrapper>& wrappers) const
+{
+  auto& leader          = wrappers.getLeader();
+  leader.mw_mem_handle_ = collection.lendResource<TWFFastDerivWrapperMultiWalkerMem>();
+}
+
+void TWFFastDerivWrapper::releaseResource(ResourceCollection& collection,
+                                          const RefVectorWithLeader<TWFFastDerivWrapper>& wrappers) const
+{
+  auto& leader = wrappers.getLeader();
+
+  // If the leader has a valid handle, take back the resource.
+  if (leader.mw_mem_handle_)
+    collection.takebackResource(leader.mw_mem_handle_);
+
+  // Now, clear the handles for all wrappers to make the operation idempotent.
+  for (auto& wref : wrappers)
+    const_cast<TWFFastDerivWrapper&>(wref.get()).mw_mem_handle_ = {};
```

Comment:
If any non-leader carries a valid resource handle, there should be a bug. I don't feel it right to blindly clear handles at this place.